kvdbclient/bigtable/attributes.py

```diff
@@ -0,0 +1,76 @@
+from typing import NamedTuple
+
+from .. import serializers
+
+
+class _AttributeType(NamedTuple):
+    key: bytes
+    family_id: str
+    serializer: serializers.Serializer
+
+
+class Attribute(_AttributeType):
+    __slots__ = ()
+    _attributes = {}
+
+    def __init__(self, **kwargs):
+        super().__init__()
+        Attribute._attributes[(kwargs["family_id"], kwargs["key"])] = self
+
+    def serialize(self, obj):
+        return self.serializer.serialize(obj)
+
+    def deserialize(self, stream):
+        return self.serializer.deserialize(stream)
+
+    @property
+    def basetype(self):
+        return self.serializer.basetype
+
+    @property
+    def index(self):
+        return int(self.key.decode("utf-8").split("_")[-1])
+
+
+class AttributeArray:
+    _attributearrays = {}
+
+    def __init__(self, pattern, family_id, serializer):
+        self._pattern = pattern
+        self._family_id = family_id
+        self._serializer = serializer
+        AttributeArray._attributearrays[(family_id, pattern)] = self
+
+        # TODO: Add missing check in `fromkey(family_id, key)` and remove this
+        #       loop (pre-creates `Attributes`, so that the inverse lookup works)
+        for i in range(20):
```

Comment:
this seems odd. Why 20? Please add that to the documentation

Reply:
@nkemnitz do remember why that was necessary?

Reply:
The 20? Just because I couldn't figure out a way to do it dynamically - the number of family IDs could change, so I used an upper bound.